examples/topic_models/lntm_mcem.py

```diff
@@ -67,42 +73,46 @@ def lntm(observed, D, K, V, eta_mean, eta_logstd):
 
     T = np.sum(X_train)
     iters = X_train.shape[0] // D
-    Eta = np.zeros((X_train.shape[0], K), dtype=np.float32)
+    Eta = np.zeros((n_chains, X_train.shape[0], K), dtype=np.float32)
     Eta_mean = np.zeros(K, dtype=np.float32)
     Eta_logstd = np.zeros(K, dtype=np.float32)
 
     # Build the computation graph
     x = tf.placeholder(tf.float32, shape=[D, V], name='x')
     eta_mean = tf.placeholder(tf.float32, shape=[K], name='eta_mean')
     eta_logstd = tf.placeholder(tf.float32, shape=[K], name='eta_logstd')
-    eta = tf.Variable(tf.zeros([D, K]), name='eta')
-    eta_ph = tf.placeholder(tf.float32, shape=[D, K], name='eta_ph')
+    eta = tf.Variable(tf.zeros([n_chains, D, K]), name='eta')
+    eta_ph = tf.placeholder(tf.float32, shape=[n_chains, D, K], name='eta_ph')
     beta = tf.Variable(tf.zeros([K, V]), name='beta')
     phi = tf.nn.softmax(beta)
-    init_eta = tf.assign(eta, tf.zeros([D, K]))
     init_eta_ph = tf.assign(eta, eta_ph)
 
     def joint_obj(observed):
-        model = lntm(observed, D, K, V, eta_mean, eta_logstd)
-        # [D], [K], [K]
+        model = lntm(observed, n_chains, D, K, V, eta_mean, eta_logstd)
+
         log_p_eta, log_p_beta = \
             model.local_log_prob(['eta', 'beta'])
 
         theta = tf.nn.softmax(observed['eta'])
         phi = tf.nn.softmax(observed['beta'])
-        pred = tf.matmul(theta, phi)
+        tiled_phi = tf.tile(tf.expand_dims(phi, 0), [n_chains, 1, 1])
```

Comment:
It should be better to reshape theta and multiple with phi, and reshape back. This saves memory.

theta = tf.reshape(theta, [-1, K])
pred = tf.matmul(theta, phi)
pred = tf.reshape(pred, [n_chains, D, -1])